### ISSUE TITLE
Webpage down?

### ISSUE THREAD

**Issue by ryland-mortlock** [ChengLabGit/RashX_optimized#2]
Hi, thank you for your wonderful tool.
I believe your web page (https://production.d2wa2wvykarq3n.amplifyapp.com/) may be down because when I upload an rds file, add my email address, and click submit, there is no response on the page.

I went ahead and implemented your process_requests() script (https://github.com/ChengLabGit/RashX_optimized/blob/main/R/process_requests.R) locally in R and got it to work. I did want to let you know that a few bugs came up associated with the switch in naming conventions for different assays/layers in Seurat v5. They were fairly straightforward to fix, but might be worth looking into. For example,
Line 576-577: `exp_mat <-
      Unknown@assays[["RNA"]]@data #pull out NORMALIZED counts from Seurat object` throws an error which can be corrected with: `exp_mat <-
    Unknown@assays[["RNA"]]@layers$data #pull out NORMALIZED counts from Seurat object`

**Comment by caeser1996** [comment]
@ryland-mortlock  can you please check again and let me know how about it?

@ryland-mortlock  can you please submit a PR with the changes that you suggest?

**Comment by ryland-mortlock** [comment]
For me, the website still doesn't work.
I will try to submit a PR soon for the `process_requests()` script, I apologize, I've been swamped lately.

**Comment by caeser1996** [comment]
strange I did checked the same and it worked let me try again.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by caeser1996** [comment]
@ryland-mortlock  Can you test it right now on the site and see if you can get a popup stating its being processed.

**Comment by caeser1996** [comment]
@ryland-mortlock  I just ran the processing there using one of the sample file in site as soon as it get completed I will share the you the same

**Comment by ryland-mortlock** [comment]
I still get no response when I click submit
<img width="1141" alt="image" src="https://github.com/ChengLabGit/RashX_optimized/assets/58830770/999e3e76-3963-4af4-9388-ed0a621e755e">
The dotted line around the RDS file makes me think maybe it's not uploading?

**Comment by ChengLabGit** [comment]
Ryland,

Can you send a copy of the RDS file you're using?
Is your sample extremely large (in terms of # of cells)?

thanks,

jeffrey
________________________________

**Comment by ryland-mortlock** [comment]
The .rds file is 1.02 GB, it has 10,464 cells and 2 assays: RNA and SCT. It is a seurat v5 object.
I will send you a link to the .rds file by email. 
Thanks.

**Comment by ChengLabGit** [comment]
Ryland,

Good to hear that the algorithm works offline with the R script.

What's SCT?  I believe the issue is that your RDS file is many fold larger than the ones we usually use, ours are < 100 MB and have 6000-8000 cells and our website doesn't have the memory to handle such a large file..  Is there a way to just generate the RDS file with RNA data only (and a smaller version).  If so, I think it will work on the rashx website.

Best,

Jeffrey
________________________________

**Comment by ryland-mortlock** [comment]
Hi Jeffrey,
SCT is the slot for normalized data using SCTransform. 
You're right, that's the issue! I just saved an RDS file of my seurat object before running SCT, with just RNA counts data (it is much smaller, 56 MB) and I get the pop-up saying "File is getting processed"
It makes sense to use the smaller dataset before normalization since your pipeline does NormalizeData itself.
Thanks for the help! @ChengLabGit @caeser1996
